airflow/models.py

```diff
@@ -1041,7 +1061,7 @@ def refresh_from_db(self, session=None, lock_for_update=False):
             self.state = ti.state
             self.start_date = ti.start_date
             self.end_date = ti.end_date
-            self.try_number = ti.try_number
+            self._try_number = ti._try_number
```

Comment:
why use _try_number here?

Reply:
If we don't use `_try_number` here we get an odd case like this:

`ti._try_number == 1` db, so `ti.try_number == 2` =>  `self._try_number = 2` so leading to `self.try_number == 3`.

Reply:
To be clear: self.try_number is fine, but we have to read from `ti._try_number`.